### PR TITLE
Just attempt creation and handle 409 errors.

### PR DESCRIPTION
**Pull request by buckett**
When attempting creation we don't check if it exists first, but just handle the case when a 409 error is generated. Although 409 should only happen on creation, we add the error handling to all tool-support methods so that they generally behave the same.

### CODE REVIEW

File: bin/index.js [hunk 1]
```diff
@@ -175,6 +175,17 @@ const checkError = (error) => {
     throw new Error('Untrusted certificate in chain')
   }
 }
+
+/**
+ * Check if tool support request is failing for a reason we know more about.
+ */
+const checkToolSupportError = (error) => {
+  if (error.response) {
+    if (error.response.status === 409) {
```
Comment (mpellicer) [inline bin/index.js:184]: Lets combine the IF

if (error.response && error.response.status === 409)

**Comment by mpellicer** [comment]
I'm not sure we should create a Canvas developer key if the clientId or the registrationId is taken?

**Comment by buckett** [comment]
There's no unique index on client ID at the moment, if we want to then we should enforce that on tool-support. At the moment you can create one and then update a different one to use the same client ID.

**Comment by buckett** [comment]
It will fail it the registration ID it taken as that will return a 409.

**Comment by buckett** [comment]
> > It will fail it the registration ID it taken as that will return a 409.
> 
> But it will create a Canvas developer key right? Don't we want to avoid that ?

Yeah, brain not working.

**Review by sebastianchristopher — APPROVED** [review]
LGTM

**Pull request merged**